pipenv/utils/resolver.py

```diff
@@ -309,15 +310,17 @@ def get_deps_from_req(
         req: Requirement,
         resolver: Optional["Resolver"] = None,
         resolve_vcs: bool = True,
-    ) -> Tuple[Set[str], Dict[str, Dict[str, Union[str, bool, List[str]]]]]:
+    ) -> Tuple[Dict[str, None], Dict[str, Dict[str, Union[str, bool, List[str]]]]]:
         from pipenv.vendor.requirementslib.models.requirements import Requirement
         from pipenv.vendor.requirementslib.models.utils import (
             _requirement_to_str_lowercase_name,
         )
         from pipenv.vendor.requirementslib.utils import is_installable_dir
 
         # TODO: this is way too complex, refactor this
-        constraints: Set[str] = set()
+        constraints: Dict[
+            str, None
+        ] = {}  # Used Dict instead of Set because Dict is ordered and is hence stable
```

Comment:
Why not a [frozenset](https://docs.python.org/3/library/stdtypes.html#frozenset)?